### ISSUE TITLE
Segmentation Fault possibly in ScatterND

### ISSUE THREAD

**Issue by truncs** [NVIDIA/TensorRT#1692]
## Description

While trying to work around empty concat node issue https://github.com/NVIDIA/TensorRT/issues/1650 I ended up with the following subgraph 
```
import torch
import torch.nn as nn


class TestModule(nn.Module):
    def __init__(self):
        super(TestModule, self).__init__()

    def forward(self, ref):
        features = torch.zeros((1, 64, 100, 160))
        target = torch.zeros((1, 32, 100, 160))
        
        idx_even = torch.arange(0, 64, 2)
        idx_odd = torch.arange(1, 64, 2)

        features[:, idx_even, :, :] = ref
        features[:, idx_odd, :, :] = target 

        features = features.contiguous()
        return features
        
```

But on using trtexec on this, it returns a segmentation fault

```
[12/27/2021-18:17:55] [V] [TRT] ScatterND_37 [ScatterND] inputs: [1 -> (1, 64, 100, 160)[FLOAT]], [48 -> (1, 32, 100, 160, 4)[INT32]], [55 -> (1, 32, 100, 160)[FLOAT]], 
[12/27/2021-18:17:55] [V] [TRT] Registering layer: 1 for ONNX node: 1
Segmentation fault (core dumped)
```

The onnx graph can be downloaded from [here.](https://drive.google.com/file/d/1eTWXWJSCzVYjPJEXQhjrH83LBw9pDJjI/view?usp=sharing)

## Environment

**TensorRT Version**: 8.2
**NVIDIA GPU**:  Xavier NX
**NVIDIA Driver Version**: N/A
**CUDA Version**:  10.2
**CUDNN Version**: 8
**Operating System**:  Ubuntu 18.04.6 LTS
**Python Version (if applicable)**:  3.6
**Tensorflow Version (if applicable)**: N/A
**PyTorch Version (if applicable)**: 1.9
**Baremetal or Container (if so, version)**:

**Comment by truncs** [comment]
If I remove one the assignments the graph in even simpler 

![segfault](https://user-images.githubusercontent.com/157234/147526126-a00bdbd4-567e-4755-945d-b2a15ba9fe32.png)

The problem seems though that ScatterND, indices are more than the source array itself 

```
ScatterND_37 [ScatterND] inputs: [1 -> (1, 64, 100, 160)[FLOAT]], [48 -> (1, 32, 100, 160, 4)[INT32]], [55 -> (1, 32, 100, 160)[FLOAT]], 
```

I am not sure how the indices was interpreted in this way since I am using a arange to generate the indices
```
        idx_even = torch.arange(0, 64, 2)
```

**Comment by truncs** [comment]
Tried with a static index set

```
        #idx_even = torch.arra
        arr = np.array([0, 2, 4, 6, 8, 10, 12, 14,
                        16, 18, 20, 22, 24, 26,
                        28, 30, 32, 34, 36, 38,
                        40, 42, 44, 46, 48, 50,
                        52, 54, 56, 58, 60, 62])
        idx = torch.as_tensor(arr)
        #idx_odd = torch.arange(1, 64, 2)

        features[:, idx, :, :] = ref
        #features[:, idx_odd, :, :] = target 

```
 which resulted in the same segmentation fault

```
[12/27/2021-20:02:48] [V] [TRT] Searching for input: 55
[12/27/2021-20:02:48] [V] [TRT] ScatterND_37 [ScatterND] inputs: [1 -> (1, 64, 100, 160)[FLOAT]], [48 -> (1, 32, 100, 160, 4)[INT32]], [55 -> (1, 32, 100, 160)[FLOAT]], 
[12/27/2021-20:02:48] [V] [TRT] Registering layer: 1 for ONNX node: 1
Segmentation fault (core dumped)
```

**Comment by truncs** [comment]
@zerollzeng @ttyio thoughts on this one?

**Comment by nvpohanh** [comment]
@truncs I can repro this using TRT 8.2.1, but it has been fixed in TRT 8.4.0/8.4.1. Could you try TRT 8.4.1 (GA)? Thanks

**Comment by truncs** [comment]
Tried it with [TensorRT v8401] and it works!
